atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/WatchedCellRanges.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.api.watch;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.transaction.api.RowReference;
+import com.palantir.dialogue.DialogueImmutablesStyle;
+import org.immutables.value.Value;
+
+public final class WatchedCellRanges {
```

Comment:
This is similar to `LockWatchReferences`, with a few key differences:
1 - the `WatchedCellRange` implementations have public visibility
2 - only supported types of `LockWatchReference` have parallels here. This makes visitors nicer to implement.
3 - the types are nicer and stronger (`TableReference` and `RowReference` vs `String` and `byte[]`).

The different implementation of `LockWatchReference` are package-private, so as to minimise how much implementation detail concerning lock watches leaks out through the API. We don't have the same concern here, because we are in `atlasdb-impl-shared`.